compiler/src/iree/compiler/Dialect/Flow/Transforms/AnnotateDispatches.cpp

```diff
@@ -80,7 +83,9 @@ static int64_t estimateLinalgExtOpCost(Operation *op) {
   // This is something like the extra log(N) factor for a sort or FFT, or
   // the amount of work done by a softmax vs a cheap elementwise on a tensor
   // of the same shape.
-  cost *= 10;
+  if (cost < kMaxCost / 10) {
+    cost *= 10;
+  }
```

Comment:
Was causing integer overflow when shape was dynamic

Reply:
How about set it to `kMaxCost` when there is an overflow? I.e.,

```suggestion
  if (cost >= kMaxCost / 10) {
    cost = kMaxCost;
  }
```

just in case if you're interested in it, the trick is called Saturation. I learned it from prior image processing work: https://en.wikipedia.org/wiki/Saturation_arithmetic

Reply:
Oh cool, that makes sense!